tools/eval/datasets/lfw.py

```diff
@@ -0,0 +1,240 @@
+    thresholds = np.arange(0, 4, 0.001)
+    val, val_std, far = calculate_val(thresholds,
+                                      embeddings1,
+                                      embeddings2,
+                                      np.asarray(actual_issame),
+                                      1e-3,
+                                      nrof_folds=nrof_folds)
+    return tpr, fpr, accuracy, val, val_std, far
+
+
+class LFW:
+    def __init__(self, root, target_size=250):
+        self.LFW_IMAGE_SIZE = 250
+
+        self.lfw_root = root
+        self.target_size = target_size
+
+        self.lfw_pairs_path = os.path.join(self.lfw_root, 'view2/pairs.txt')
+        self.image_path_pattern = os.path.join(self.lfw_root, 'lfw', '{person_name}', '{image_name}')
+
+        self.lfw_image_paths, self.id_list = self.load_pairs()
+
+    @property
+    def name(self):
+        return 'LFW'
+
+    def __len__(self):
+        return len(self.lfw_image_paths)
+
+    @property
+    def ids(self):
+        return self.id_list
+
+    def load_pairs(self):
+        image_paths = []
+        id_list = []
+        with open(self.lfw_pairs_path, 'r') as f:
+            for line in f.readlines()[1:]:
+                line = line.strip().split()
+                if len(line) == 3:
+                    person_name = line[0]
+                    image1_name = '{}_{:04d}.jpg'.format(person_name, int(line[1]))
+                    image2_name = '{}_{:04d}.jpg'.format(person_name, int(line[2]))
+                    image_paths += [
+                        self.image_path_pattern.format(person_name=person_name, image_name=image1_name),
+                        self.image_path_pattern.format(person_name=person_name, image_name=image2_name)
+                    ]
+                    id_list.append(True)
+                elif len(line) == 4:
+                    person1_name = line[0]
+                    image1_name = '{}_{:04d}.jpg'.format(person1_name, int(line[1]))
+                    person2_name = line[2]
+                    image2_name = '{}_{:04d}.jpg'.format(person2_name, int(line[3]))
+                    image_paths += [
+                        self.image_path_pattern.format(person_name=person1_name, image_name=image1_name),
+                        self.image_path_pattern.format(person_name=person2_name, image_name=image2_name)
+                    ]
+                    id_list.append(False)
+        return image_paths, id_list
+
+    def __getitem__(self, key):
+        img = cv.imread(self.lfw_image_paths[key])
+        if self.target_size != self.LFW_IMAGE_SIZE:
+            img = cv.resize(img, (self.target_size, self.target_size))
+        return img
+
+    def eval(self, model):
+        ids = self.ids
+        embeddings = np.zeros(shape=(len(self), 128))
+        for idx, img in tqdm(enumerate(self), desc="Evaluating {} with {} val set".format(model.name, self.name)):
+            embedding = model.infer(img)
```

Comment:
The image should be aligned before being fed into the model. This can be the problem of low accuracy.

To get aligned image, you need to get the bounding box of the face using YuNet. For simplicity, you can start with treating the bounding box of highest score as the only face in the image. So I suggest you:
1. Validate whether alignment using YuNet improves the accuracy.
2. If it does, store those boxes in a constant in this file (or a separate .npy file) and load it when requires to save time.

Reply:
Thanks! After using YuNet to get bbox, the accuracy is 97.92%. this is still not 99.6% as described in the documentation. I will try to solve this problem.

Reply:
The difference can be YuNet failing to detect a face in some images. You can have a check on the images with low score faces or no face.

Reply:
It may not be the problem. Each image is detected with a face and the selected bbox scores are all above 0.8.

Reply:
Sometimes it may fail even with score 0.8. Please, take a look at those faces with score lower than 0.9.